### PR TITLE
Better error dialogs

### PR DESCRIPTION
**Pull request by MrTroble**

### CODE REVIEW

File: src/main/java/com/troblecodings/launcher/util/ConnectionUtil.java [hunk 1]
```diff
@@ -1,121 +1,172 @@
...
+					Launcher.INSTANCEL.setPart(new ErrorPart(Launcher.INSTANCEL.getPart(), "Error verifying " + Paths.get(name).getFileName().toString(), "The file failed to download correctly after 3 tries!"));
+					break;
+				}
+				ConnectionUtil.download(url, name);
+				times++;
+			}
+		});
 	}
 
 }
```
Comment (Der-Zauberer) [inline src/main/java/com/troblecodings/launcher/util/ConnectionUtil.java:172]: I can't find out why, but the code throws exceptions, if the modpack is started after using the reset feature. This happend with different files, but all with the same exception.
```
java.nio.file.NoSuchFileException: C:\Users\andso\AppData\Roaming\gir\libraries\net\sf\jopt-simple\jopt-simple\5.0.3\jopt-simple-5.0.3.jar.tmp
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:318)
	at sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:287)
	at java.nio.file.Files.move(Files.java:1395)
	at com.troblecodings.launcher.util.ConnectionUtil.download(ConnectionUtil.java:113)
	at com.troblecodings.launcher.util.ConnectionUtil.download(ConnectionUtil.java:90)
	at com.troblecodings.launcher.util.ConnectionUtil.lambda$0(ConnectionUtil.java:166)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Review by Der-Zauberer — CHANGES_REQUESTED** [review]
ConnectionUtil needs to be fixed because of exceptions on launching the modpack after using the reset feature.

File: src/main/java/com/troblecodings/launcher/util/ConnectionUtil.java (hunk 1 above)
Comment (MrTroble) [reply src/main/java/com/troblecodings/launcher/util/ConnectionUtil.java:172]: Very good testing 👍 I will take a look into it

Comment (MrTroble) [reply src/main/java/com/troblecodings/launcher/util/ConnectionUtil.java:172]: This is an issue with the executor pool ... turns out parallel downloading isn't the greatest of all ideas. I have removed the feature. Thanks for checking.

**Review by Der-Zauberer — APPROVED** [review]

**Pull request merged**